20-ensemble_models.Rmd

```diff
@@ -7,13 +7,118 @@
 - Fit the **member models** using `fit_members()`.
 - Test the results of an ensemble model.
 
-## Slide 1 Title
+## Ensembling
 
-Put the content of your slide here.
+- Aggregating predictions of multiple models together to make one prediction.
+- We've already seen some ensembling within 1 type of model:
+  - Random forest
+  - Bagging & boosting
+  
+- **Model stacking** allows for aggregating many different types of models (lm + rf + svm, ...), to assemble a new model, which generates a new prediction, informed by its *members*.
+
+## Ensembling with `stacks`!
+
+![](https://raw.githubusercontent.com/tidymodels/stacks/main/man/figures/logo.png){width=25%}
+
+https://stacks.tidymodels.org/articles/basics.html
+
+Model stacking process:
+
+1. Define some models, using all the knowledge we have from this book. We call them *candidate models/members*.
+2. Initialize the ensemble with `stacks::stacks()`, and add the members to it, using `stacks::add_members()`
+3. Blend the predictions of the members, using `stacks::blend_predictions()` (linear combination of each member's predictions)
+4. Now that we know how to blend members, `fit` the members one more time on the whole training set, and `predict` on testing set.
+
+## Define some models
+
+- Ensembles are formed from model definitions, which are `workflow`s that contain model recipe & spec (that has been tuned or fit_resample'd).
+- The book recommends the racing tuning method.
+
+![](images/20_model-definitions.PNG)
+
+- You’ll need to save the assessment set predictions and workflow utilized in your `tune_grid()`, or `fit_resamples()` objects by setting the control arguments `save_pred = TRUE` and `save_workflow = TRUE`. 
+
+- If a model has hyperparameters, then you'd be creating multiple candidate models.
+
+![](images/20_model-definitions-hyperparameters.PNG)
+
+## Initialize and add members to stack.
+
+- Data stacks are tibbles (with some extra attributes) that contain the response value as the first column, and the assessment set predictions for each candidate ensemble member.
+
+![](images/20_model-stack.PNG)
+
+```{r eval=FALSE}
```

Comment:
Please give all r code blocks a unique title, like "20-tree-frogs". That makes it much easier to diagnose issues in the future if something fails.

Reply:
arghh! Thanks for pointing it out :) Should be good!